sda-download/dev_utils/config.yaml

```diff
@@ -4,7 +4,9 @@ app:
   serverkey: "./dev_utils/certs/download-key.pem"
   port: "8443"
   middleware: "default"
-  serveUnencryptedData: true
+  c4gh:
+    passphrase: ""
+    privateKeyPath:  ""
```

Comment:
What is the point of this addition?
The crypt4gh key stuff are declared further down in the config file and we only need one.

Reply:
I would expect that the internal key that `download` uses is not the same as the one that `reencrypt` uses (the latter being the archive's main key).

Reply:
Also, I thought this addition was your suggestion? https://github.com/neicnordic/sensitive-data-archive/pull/1163#discussion_r1883523274

Reply:
The config file for `sda-download` is not unified with the one for the rest of the sda-stack and if a single file is used to start all services you will get into trouble.

The config file in this folder should be for sda-download and noting else.

Reply:
> I would expect that the internal key that `download` uses is not the same as the one that `reencrypt` uses (the latter being the archive's main key).

It should not be the same, but the config file for sda-download should only have one c4gh entry until it is merged into the `sda` code structure.

Reply:
I am not following,  the config file here is used for sda-download's integration tests and there you need both sda-download and reencrypt c4gh keys. Please elaborate.

Reply:
Yes, and that is the root of the problem, there is one config file that is used to configure two services that do not have a unified code to parse the configuration.

Reply:
yes, but that was a problem all along and it will be as long as we don't merge `sda-download` to the stack. I don't see how this is relevant here specifically

Reply:
The code around `sda-download` (integration tests and what lies in the dev_utils directory) is a mess. My point is that fixing this is not part of this PR.

Reply:
Fixing all the tests is out of the scope, but you can take one step in the right direction by moving out everything related to the re-encrypt service from the conifig.yaml. Either in to a separate config file or into the compose file

Reply:
I find it quite hard to follow this, so I agree that better documentation would be nice :sweat_smile: . To me, it sounds like it's time for a new issue, and only going for a minimal working solution (regarding the config file) in this PR.

Reply:
As a resolve to this, I opened Issue #1209,  feel free to refine the description of the issue :smile: .